pages/mesosphere/dcos/2.2/storage/external-storage/csi/index.md

```diff
@@ -0,0 +1,119 @@
+-  `external.pluginName`: The name of the CSI plugin which will mount this volume. This is specified in the CSI plugin configuration on the agent; for the Portworx service, this should be `pxd.portworx.com`. For NFS volumes, this should be `nfs.csi.k8s.io`. For other drivers, see the instructions on driver installation below.
+-  `external.options.accessMode`: The [CSI access mode][2] to use when mounting the volume. The `XXXX_WRITER` modes will be mounted in read-write, while the `READER_ONLY` modes will be mounted read-only.
+-  `external.options.nodeStageSecret` and `external.options.nodePublishSecret`: The names of secrets in the DC/OS secret store which contain the username/password to use with the storage backend when staging and publishing this volume. "Staging" and "publishing" refer to steps in the CSI volume mounting process.
+-  `external.options.accessType.fsType`: The type of filesystem to be used on this volume, i.e. "xfs", "ext4", etc.
+-  `external.options.accessType.mountFlags`: Flags to be used when mounting the volume on the host.
+-  `external.options.volumeContext`: In some cases, the provisioning of a volume results in the creation of volume context metadata which must be passed as key-value pairs in this field.
+-  Create multiple volumes by adding additional items in the `container.volumes` array.
+-  You cannot change volume parameters after you create the application.
+-  Marathon will not launch apps with external volumes if `upgradeStrategy.minimumHealthCapacity` is greater than 0.5, or if `upgradeStrategy.maximumOverCapacity` does not equal 0.
+
+Some of the options above map directly onto fields in the [`NodeStageVolume`][3] and [`NodePublishVolume`][4] calls in the CSI spec; some useful information regarding those parameters can be found in the CSI docs.
+
+## Create an application from the DC/OS UI
+
+1. Click the **Services** tab, then **RUN A SERVICE**.
+1. Click **Volumes** and enter your Volume ID, Plugin Name, and Container Path.
+1. Click **Deploy**.
+
+# Scaling your app
+
+Marathon apps which use CSI volumes can currently only be scaled to a single instance. This constraint will be relaxed in the future.
+
+# Using 3rd party CSI drivers
+
+If you want to use a third party CSI driver which has not yet integrated with DC/OS, you can do the following to accomplish this:
+
+1. Install the CSI driver as a Marathon app. To integrate with DC/OS, the CSI driver must expose a Unix domain socket which DC/OS can connect to in order to make CSI calls. Install the app with a `hostname:UNIQUE` constraint so that only one instance is installed on each node. To make this storage backend available on all nodes, install the driver on all private agents by setting the number of instances equal to the number of private agents in the cluster.
```

Comment:
Minor suggestion, can we point to the Portworx drivers as an example of a CSI Unix Domain Socket?